File: 35_搜索插入位置/searchInsert.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        length = len(nums)
        left = 0
        right = length - 1
        while right - left >= 0:  # 这样mid是可以取到两端指针的
            mid = (right + left) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        return left

File: 35_搜索插入位置/searchInsert.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    const n = nums.length
    let left = 0
    let right = n - 1

    while(left <= right){
        let mid = Math.floor(left + (right - left) / 2)
        if(nums[mid] < target){
            left = mid + 1
        }else if(nums[mid] > target){
            right = mid - 1
        }else if(nums[mid] === target){
            right = mid - 1
        }
    }

    return left
};

// console.log(searchInsert([1,3,5,6], 2))
// console.log(searchInsert([1,3,5,6], 7))